zebra-state/src/service/finalized_state.rs

```diff
@@ -171,21 +163,23 @@ impl FinalizedState {
                 finalized.height.0 as f64,
             );
             metrics::counter!("zcash.chain.verified.block.total", 1);
-
-            Ok(finalized)
         } else {
             metrics::counter!("state.checkpoint.error.block.count", 1);
             metrics::gauge!(
                 "state.checkpoint.error.block.height",
                 finalized.height.0 as f64,
             );
-
-            Err(())
         };
 
-        let _ = rsp_tx.send(result.map_err(Into::into));
+        // Some io errors can't be cloned, so we format them instead.
+        let owned_result = result
+            .as_ref()
+            .map(|_hash| finalized)
+            .map_err(|error| format!("{:?}", error).into());
+
+        let _ = rsp_tx.send(result);
```

Comment:
Could we swap these two statements, and return the adjusted result directly?

Reply:
`send()` takes ownership of the Result, and the error isn't cloneable, so we have to format the error before we send it to two different places.

Reply:
This should have been obvious to me. Sorry.

Reply:
> This should have been obvious to me. Sorry.

No worries, it's not obvious in the code, it trips me up all the time!